packages/SwingSet/src/liveslots/liveslots.js

```diff
@@ -957,6 +983,7 @@ function build(
           // if we didn't forbid promises, we'd need to
           // maybeExportPromise() here
           forbidPromises(serArgs);
+          assertAcceptableSyscallCapdataSize(serArgs);
```

Comment:
As above, we should do the checks before the `retainExportedVref`. In fact, please move the `forbidPromises()` test (but not the commented-out `maybeExportPromise()` bit) up a line too, all the tests should happen before we commit to state changes like `retainExportedVref`, to minimize the potential for leaks.